lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1823,162 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    id_searches, type_searches, checkin_searches, searches = [], [], [], []
+
+    # Group search terms by what's being searched for
+    terms.each do |term|
+      case term.split(":").first
+      when "session_id"
+        id_searches << term
+      when "session_type"
+        type_searches << term
+      when "last_checkin"
+        checkin_searches << term
+      else
+        print_error("Please provide valid search term. Given: #{term.split(":").first}")
+        return
+      end
+    end
+
+    # Make sure user is properly querying checkin
+    unless checkin_searches.length() < 2
+      unless validate_checkin_searches(checkin_searches)
+        return
+      end
+    end
+
+    # Group results by search term - OR filters
+    [id_searches, type_searches].each do |terms|
+      unless terms.empty?
+        id_matches = {}
+        terms.each do |term|
+          matches = filter_sessions_by_search(term)
+          return unless matches
+          id_matches = id_matches.merge(matches)
+        end
+        searches << id_matches
+      end
+    end
+
+    # Retrieve checkin search results. AND filter with a max length of 2
+    unless checkin_searches.empty?
+      checkin_matches = {}
+      checkin_matches = filter_sessions_by_search(checkin_searches.first)
+      if checkin_searches[1]
+        matches = filter_sessions_by_search(checkin_searches[1])
+        checkin_matches = checkin_matches.select{ |session_id, session| matches[session_id] == session }
+      end
+      searches << checkin_matches
+    end
```

Comment:
```
checkin_matches = {}
checkin_searches.each do | search_query |
 ...
end
```
you should just be able to loop over the contents and achieve the same thing you're trying to do here

Reply:
I'm a little confused what a loop would add here since checkin_searches can't be more than two elements and all we're trying to do is get the intersection of the two if there are two, can you elaborate?

Reply:
uhhh yea I was thinking it could look a bit tidier and maybe more future proof but don't worry about it sure